File: safelima/lib/models/policestations.dart
class PoliceStation {
  final int id;
  final String nombre;
  final String? direccion;
  final String? telefono;
  final double latitud;
  final double longitud;
  final String? distrito;
  final DateTime? fechaRegistro;
  final double? distanciaKm;

  PoliceStation({
    required this.id,
    required this.nombre,
    this.direccion,
    this.telefono,
    required this.latitud,
    required this.longitud,
    this.distrito,
    this.fechaRegistro,
    this.distanciaKm,
  });

  factory PoliceStation.fromJson(Map<String, dynamic> json) {
    return PoliceStation(
      id: json['id'],
      nombre: json['nombre'],
      direccion: json['direccion'],
      telefono: json['telefono'],
      latitud: (json['latitud'] as num).toDouble(),
      longitud: (json['longitud'] as num).toDouble(),
      distrito: json['distrito'],
      fechaRegistro: json['fecha_registro'] != null
          ? DateTime.tryParse(json['fecha_registro'])
          : null,
      distanciaKm: json['distancia_km'] != null
          ? (json['distancia_km'] as num).toDouble()
          : null,
    );
  }
}

File: safelima_app/app/models/police_stations.py
from sqlalchemy import Column, Integer, String, Text, Numeric, DateTime, ForeignKey, UniqueConstraint
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database.database import Base

class PoliceStation(Base):
    __tablename__ = "police_stations"
    
    id = Column(Integer, primary_key=True, index=True)
    nombre = Column(String(100), nullable=False)
    direccion = Column(Text)
    telefono = Column(String(20))
    latitud = Column(Numeric(10, 6), nullable=False)
    longitud = Column(Numeric(10, 6), nullable=False)
    distrito = Column(String(50))
    fecha_registro = Column(DateTime, default=datetime.utcnow)